### PR TITLE
Fix Synthesizable Unittests

### PR DESCRIPTION
**Pull request by davidbiancolin**
I'd like to include these in CI but they've bit rotted a bit through the last couple of RC bumps.

### CODE REVIEW

File: sim/midas/src/main/scala/midas/SynthUnitTests.scala [hunk 1]
```diff
@@ -23,7 +23,8 @@ class WithAllUnitTests extends Config((site, here, up) => {
       Module(new ReadyValidChannelUnitTest(timeout = timeout)),
       Module(new CounterTableUnitTest),
       Module(new LatencyHistogramUnitTest),
-      Module(new AddressRangeCounterUnitTest))
```
Comment (zhemao) [inline sim/midas/src/main/scala/midas/SynthUnitTests.scala:26]: Is there any reason to change this beyond aesthetics?

File: sim/midas/src/main/scala/midas/core/Channel.scala [hunk 2]
```diff
@@ -241,6 +241,8 @@ class ReadyValidChannelUnitTest(
 
   val dut = Module(new ReadyValidChannel(payloadType))
   val reference = Module(new ShiftQueue(payloadType, queueDepth))
+  val referenceReset = Wire(Bool())
+  reference.reset := referenceReset
```
Comment (zhemao) [inline sim/midas/src/main/scala/midas/core/Channel.scala:245]: Why does this need to be a wire?

File: sim/midas/src/main/scala/midas/SynthUnitTests.scala (hunk 1 above)
Comment (davidbiancolin) [reply sim/midas/src/main/scala/midas/SynthUnitTests.scala:26]: i was commenting lines out to figure out which test was causing FIRRTL to croak, this made that more brainless.

File: sim/midas/src/main/scala/midas/core/Channel.scala (hunk 2 above)
Comment (davidbiancolin) [reply sim/midas/src/main/scala/midas/core/Channel.scala:245]: reference.reset is an abstract Reset. For the LIBDN tester hardware to generate correctly it needs to be a concrete reset -- specifically a Bool in this case.  Using the wire here both gives us a reference to drive from the LIBDN tester the and forces a particular concrete type on to the port.

**Pull request merged**